code/processes/tailer.q

```diff
@@ -0,0 +1,57 @@
+\d .wdb
+.proc.loadf [getenv[`KDBCODE],"/processes/wdb.q"]
+hdbsettings:(`compression`hdbdir`taildir)!(compression;hdbdir;getenv`KDBTAIL)
+
+\d .tailer
+trtype:`$"tr_",last "_" vs string .proc.proctype                           /-extract wdb proc segname and append to "tr_"
+tailreadertypes:trtype
+
+/- evaluate contents of d dictionary asynchronously
+/- flush tailreader handles after timeout
+flushtailreload:{
+  if[not @[value;`.tailer.tailreloadcomplete;0b];
+   @[{neg[x]"";neg[x][]};;()] each key d;
+   .lg.o[`tail;"tailreload is now complete"];
+   .tailer.tailreloadcomplete:1b];
+  };
+
+dotailreload:{[pt]
+  /-send reload request to tailreaders
+  .tailer.tailreloadcomplete:0b;
+  .wdb.getprocs[;pt].tailer.tailreadertypes;
+  if[eodwaittime>0;
+    .timer.one[.wdb.timeouttime:.proc.cp[]+.wdb.eodwaittime;(value;".tailer.flushtailreload[]");"release all tailreaders as timer has expired";0b];
+  ];
+  };
+
+\d .wdb
+reloadproc:{[h;d;ptype;reloadlist]
+        .wdb.countreload:count[raze .servers.getservers[`proctype;;()!();1b;0b] each reloadlist];
+        $[eodwaittime>0;
+                {[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]}[({@[`. `reload;x;()]; (neg .z.w)(`.wdb.handler;1b); (neg .z.w)[]};d);h;ptype];
+                @[h;(`reload;d);{[ptype;e] .lg.e[`reloadproc;"failed to reload the ",string[ptype],".  The error was : ",e]}[ptype]]
+        ];
+        .lg.o[`reload;"the ",string[ptype]," has been successfully reloaded"];
+        }
+
+getprocs:{[x;y]
+        a:exec (w!x) from .servers.getservers[`proctype;x;()!();1b;0b];
+        /-exit if no valid handle
+        if[0=count a; .lg.e[`connection;"no connection to the ",(string x)," could be established... failed to reload ",string x];:()];
+        .lg.o[`connection;"connection to the ", (string x)," has been located"];
+        /-send message along each handle a
+        reloadproc[;y;value a;x] each key a;
+        }
+
+.servers.CONNECTIONS:(distinct .servers.CONNECTIONS,.wdb.hdbtypes,.wdb.rdbtypes,.wdb.gatewaytypes,.wdb.tickerplanttypes,.wdb.sorttypes,.wdb.sortworkertypes,.tailer.tailreadertypes) except `
+
+\d .
+
+/- initialise datastripe
+if[.ds.datastripe;
+  .lg.o[`dsinit;"datastripe on: initialising datastripe"];
+  initdatastripe[]];
+
+/- create HDB sym file and taildir symlink
+if[.ds.datastripe;
+  .ds.symlink[]];
```

Comment:
conditions are the same, why not just combine these?

personal preference but I also prefer to close `if` statements with the closing `]` on it's own line, just to make it super clear at a glance where it ends

```suggestion
/- initialise datastripe
if[.ds.datastripe;
  .lg.o[`dsinit;"datastripe on: initialising datastripe"];
  initdatastripe[];
  /- create HDB sym file and taildir symlink
  .ds.symlink[];
 ];
```

final point about this - is it valid to be running a tailer process with `.ds.datastripe` false? My understanding is that the tailer only makes sense with datastriping enabled, so should it instead be a case of throwing an error here if tailer is run and datastriping is *not* enabled?